File: validador.js
function validarCPF(cpf) {
  cpf = cpf.replace(/[^\d]+/g,''); // Remove caracteres não numéricos

  if(cpf == '') return false;

  // Valida CPF
  if (cpf.length != 11 ||
      cpf == "00000000000" ||
      cpf == "11111111111" ||
      cpf == "22222222222" ||
      cpf == "33333333333" ||
      cpf == "44444444444" ||
      cpf == "55555555555" ||
      cpf == "66666666666" ||
      cpf == "77777777777" ||
      cpf == "88888888888" ||
      cpf == "99999999999")
      return false;

  var soma = 0;
  for (var i = 0; i < 9; i++) {
      soma += parseInt(cpf.charAt(i)) * (10 - i);
  }

  var resto = 11 - (soma % 11);
  if (resto == 10 || resto == 11) {
      resto = 0;
  }
  if (resto != parseInt(cpf.charAt(9))) {
      return false;
  }

  soma = 0;
  for (var i = 0; i < 10; i++) {
      soma += parseInt(cpf.charAt(i)) * (11 - i);
  }

  resto = 11 - (soma % 11);
  if (resto == 10 || resto == 11) {
      resto = 0;
  }
  if (resto != parseInt(cpf.charAt(10))) {
      return false;
  }

  return true;
}

function validarEmail(email) {
  var re = /\S+@\S+\.\S+/;
  return re.test(email);
}

function validarTelefone(telefone) {
  telefone = telefone.replace(/[^\d]+/g,''); // Remove caracteres não numéricos

  if(telefone == '') return false;

  // Valida telefone
  if (telefone.length < 10 || telefone.length > 11) {
    return false;
  }

  return true;
}

File: src/validador.py
# Conjunto de REGEX para validação de entradas.
import re

email_regex = re.compile(r'([A-Za-z0-9]+[.-_])*[A-Za-z0-9]+@[A-Za-z0-9-]+(\.[A-Z|a-z]{2,})+')
cep_regex = r'^\d{5}-?\d{3}$'
cpf_regex = r'^\d{3}\.?\d{3}\.?\d{3}-?\d{2}$'
telefone_regex = r'^\(?([0-9]{2})\)?[-. ]?([0-9]{4})[-. ]?([0-9]{4})$'


def validar_email(email):
    if re.fullmatch(email_regex, email):
      print("Email valido")
      return True
    else:
      print("Email invalido")
      return False
  
def validar_cep(cep):
  if re.match(cep_regex, cep):
    print("CEP válido!")
    return True
  else:
    print("CEP inválido.")
    return False
  
def validar_cpf(cpf):
  if re.match(cpf_regex, cpf):
    print("CPF válido!")
    return True
  else:
    print("CPF inválido.")
    return False
  
def validar_telefone(telefone):
  if re.match(telefone_regex, telefone):
    print("Número de telefone válido!")
    return True
  else:
    print("Número de telefone inválido.")
    return False
 